### PR TITLE
Read ruby version in your gemfile

### PR DESCRIPTION
**Pull request by andycroll**

### CODE REVIEW

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+category: ruby
+image:
+  base: '2021/read-ruby-version-in-your-gemfile'
+  alt: 'Falling white dice'
+  credit:  Riho Kroll
+  source: "https://unsplash.com/photos/m4sGYaHYN5o"
+
+---
+
+Previously I recommended [using a loose version ruby version constraint in your Gemfile](/ruby/use-loose-ruby-versioning-in-your-gemfile/). This is still a valuable technique, but here’s another useful variation that's good for most of us, most of the time.
```
Comment (nodunayo) [inline _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]: *that _works well_ for most of us, most of the time ?

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Hat tip to [Emma](https://twitter.com/EmmaBeynon) for the this one-liner.
+
+Ruby version managers (such as [chruby](https://github.com/postmodern/chruby)) ensure an application uses a specific version of Ruby by looking for a [`.ruby-version` file](https://gist.github.com/fnichol/1912050) in the root directory of each app.
+
+The file specifies the required version of Ruby for this application and the manager automatically switches the environment to use the specified version.
```
Comment (nodunayo) [inline _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]: *_the_ application

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+[The version can also be specified in your Gemfile](https://bundler.io/gemfile_ruby.html) using bundler, which is more often used in a deployed environment.
```
Comment (nodunayo) [inline _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]: Are you saying 'bundler' is more often used in a deployed env, or this technique is?

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# Gemfile
+ruby "2.7.2"
+```
+
+```ruby
+# .ruby-version
+3.0.0
+```
+
+This results in an error, as bundler checks the current version of Ruby.
```
Comment (nodunayo) [inline _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]: *bundler checks the current version of Ruby against what's specified in your Gemfile... ?

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+```ruby
+# .ruby-version
+3.0.0
+```
+
+
+## Why?
+
+This means that when you upgrade Ruby you can make the change once, in the `.ruby-version` file.
```
Comment (nodunayo) [inline _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]: *you only need to update one place, the .ruby-version file.

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+Most of us, most of the time, are happy to specify a version of Ruby for all environments and this is a cool micro-improvement that reminds us that the `Gemfile` is just another ruby script!
+
+
+## Why not?
+
+There are still services where the availability of new Ruby versions are delayed so the prevous [loose ruby version constraint](/ruby/use-loose-ruby-versioning-in-your-gemfile/) might be a better fit for your project.
```
Comment (nodunayo) [inline _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]: *loose ruby version (constraint) technique ?

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md (hunk 1 above)
Comment (andycroll) [reply _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]:
```suggestion
Previously I recommended [using a loose version ruby version constraint in your Gemfile](/ruby/use-loose-ruby-versioning-in-your-gemfile/). This is still a valuable technique, but here’s another useful variation that's works well for most of us, most of the time.
```

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md (hunk 2 above)
Comment (andycroll) [reply _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]:
```suggestion
The file specifies the required version of Ruby for the application and the manager automatically switches the environment to use the specified version.
```

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md (hunk 3 above)
Comment (andycroll) [reply _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]:
```suggestion
[The version can also be specified in your Gemfile](https://bundler.io/gemfile_ruby.html) using bundler, which is more often used to define the version of Ruby to use in a deployed application.
```

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md (hunk 4 above)
Comment (andycroll) [reply _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]:
```suggestion
This results in an error, as bundler checks the current version of Ruby specified in the `Gemfile`.
```

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md (hunk 6 above)
Comment (andycroll) [reply _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]:
```suggestion
There are still services where the availability of new Ruby versions are delayed so the prevous [loose ruby version technique](/ruby/use-loose-ruby-versioning-in-your-gemfile/) might be a better fit for your project.
```

File: _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md (hunk 5 above)
Comment (andycroll) [reply _posts/2021/2021-01-11-read-ruby-version-in-your-gemfile.md]:
```suggestion
This means that when you upgrade Ruby you only need to update in one place: the `.ruby-version` file.
```

**Pull request merged**